core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/DefaultAwsV4HttpSigner.java

```diff
@@ -270,11 +293,11 @@ private static CompletableFuture<AsyncSignedRequest> doSign(AsyncSignRequest<? e
     }
 
     private static Duration validateExpirationDuration(Duration expirationDuration) {
-        if (expirationDuration.compareTo(PRESIGN_URL_MAX_EXPIRATION_DURATION) > 0) {
+        if (expirationDuration.compareTo(Duration.ZERO) < 1 || expirationDuration.compareTo(PRESIGN_URL_MAX_EXPIRATION_DURATION) > 0) {
             throw new IllegalArgumentException(
-                "Requests that are pre-signed by SigV4 algorithm are valid for at most 7" +
+                "Requests that are pre-signed by SigV4 algorithm are valid for at least 1 second and at most 7" +
```

Comment:
can be less than 1 second right? just non-zero. Just `new IllegalArgumentException("EXPIRATION_DURATION must be non-zero and at most 7 days")`?

Reply:
Minimum value is 1 second per https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html

Reply:
Ah ok. Then the check should be `expirationDuration.compareTo(Duration.ofSeconds(1)) < 0`?

Also, what's our rounding behavior? Maybe a test should assert the expected rounding behavior.

Reply:
I have some follow ups on the rounding and edge cases around 0 < duration < 1 second , but let's take that separately, as I don't want to block this PR that fixes broken tests on it.